clang/lib/Analysis/FlowSensitive/DataflowEnvironment.cpp

```diff
@@ -1104,12 +1104,22 @@ RecordStorageLocation *getBaseObjectLocation(const MemberExpr &ME,
   return Env.get<RecordStorageLocation>(*Base);
 }
 
-std::vector<FieldDecl *> getFieldsForInitListExpr(const RecordDecl *RD) {
+std::vector<const FieldDecl *>
+getFieldsForInitListExpr(const InitListExpr *InitList) {
+  const RecordDecl *RD = InitList->getType()->getAsRecordDecl();
+  assert(RD != nullptr);
+
+  std::vector<const FieldDecl *> Fields;
```

Comment:
I wonder if a `reserve` call for this vector would be beneficial. Even if we do not know the exact number of fields we want to insert, probably relatively few of them are unnamed bit fields in most code bases.

Reply:
Makes sense. Unfortunately, `RecordDecl` doesn't have a way (that I know of) to query the number of fields. There's no `getNumFields()` or similar, and `field_end()-field_begin()` also doesn't work (because the iterator doesn't support `operator-`. So we'd have to do a complete iteration through the fields to determine the number, and at that point, I think we're getting into diminishing returns.